src/sql/AtlasSQLite.cpp

```diff
@@ -0,0 +1,189 @@
+#include "AtlasSQLite.h"
+namespace {
+
+	const char *sqlCreateTable = "CREATE TABLE IF NOT EXISTS IdentityPSK("  \
+        						 "Id INTEGER PRIMARY KEY AUTOINCREMENT," \
+        						 "Identity TEXT NOT NULL UNIQUE," \
+        						 "PSK TEXT NOT NULL );";
+	const char *sqlInsert = "INSERT INTO IdentityPSK(Identity,PSK) VALUES(?,?);";// SELECT(?,?) WHERE NOT EXISTS (SELECT Identity,PSK FROM IdentityPSK WHERE Identity=?);";
+	const char *sqlUpdate = "UPDATE IdentityPSK SET PSK=? WHERE Identity=?;";
+	const char *sqlCheckIdentity = "SELECT 1 FROM IdentityPSK WHERE Identity=?;";
+	const char *sqlSelect = "SELECT PSK FROM IdentityPSK WHERE Identity=?;";
+
+}
+
+namespace atlas {
+
+AtlasSQLite::AtlasSQLite()
+{
+    isConnected_ = false;
+}
+
+AtlasSQLite::~AtlasSQLite()
+{
+    closeConnection();
+}
+
+void AtlasSQLite::closeConnection()
+{
+	if(pCon_)
+		sqlite3_close(pCon_);
+	ATLAS_LOGGER_DEBUG("Connection closed\n");
+}
+
+bool AtlasSQLite::isConnected()
+{
+    return isConnected_;
+}
+
+bool AtlasSQLite::openConnection(const std::string &databaseName, const std::string &databaseDir)
+{
+ 	isConnected_ = true;
+
+	std::string path = databaseDir + databaseName;
+
+	if(sqlite3_open(path.c_str(), &pCon_))
+	{
+		std::string err = "Can't open database: ";
+		err = err.append(sqlite3_errmsg(pCon_));
+		ATLAS_LOGGER_ERROR(err);
+		isConnected_ = false;
+	}
+
+	/* Create table 'IdentityPSK' if not exists */
+	if(sqlite3_exec(pCon_, sqlCreateTable, NULL, 0, &lastError_) != SQLITE_OK )
+	{
+		std::string err = "Can't create table: ";
+		err = err.append(lastError_);
+		ATLAS_LOGGER_ERROR(err);
+		sqlite3_free(lastError_);
+		isConnected_ = false;
+	} 
+	
+	ATLAS_LOGGER_DEBUG("Connection opened\n");
+
+	return isConnected_;
+}
+uint8_t AtlasSQLite::insert(const std::string &identity, const  std::string &psk)
+{
+    if(!isConnected())
+		return -1;
+
+	sqlite3_stmt *stmt;
```

Comment:
Shoudn't stmt object be destroyed using sqlite3_finalize? Isn't this a leak?

Reply:
Done. Checked for memory leaks with Valgrind(1 definitely lost block - reuse of memory in sqlite3_prepare_v2 function).